src/api/spec/models/user_spec.rb

```diff
@@ -804,6 +744,46 @@
     end
   end
 
+  describe '#authenticate' do
+    context 'when using a deprecated password' do
+      let(:user) { create(:user_deprecated_password) }
+
+      context 'conversation of deprecated password' do
+        before do
+          user.authenticate('opensuse')
+        end
+
+        it 'converts the password to bcrypt' do
+          expect { BCrypt::Password.new(user.password_digest) }.to_not raise_error
```

Comment:
What about:
``expect { BCrypt::Password.new(digest).is_password?("unencrypted_password") }.to be_truthy`` ?

The problem what I have with this test is that it implies that I know that ``BCrypt::Password.new(user.password_digest)`` will raise an error if the input is not a valid bcrypt hash. Also you only check that there is any error raised but actually we're looking for ``BCrypt::Errors::InvalidHash: invalid hash``, right?

Reply:
* Agreed to the first part. Already changed it.
* The intention behind the "not to check for an exact exception" is basically because you don't want other exceptions to pass by